tools/scoring-api.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
+#
+# Usage:   ./tools/scoring-api.py MODEL_NAME DATASET.CSV WEIGHTSFILE.H5 [PORT] [PARAM=VALUE]...
+# Example: ./tools/scoring-api.py rnn data/anssel/yodaqa/full-dataset.csv weights/weights-rnn-38a84de439de6337-bestval.h5 5001 loss=\"binary_crossentropy\"
+# This script loads keras model and pre-trained weights for sentence/property selection
+# and starts REST API for scoring question - sentence pairs.
+# The script listens on given or default (5000) port and accepts JSON in following format:
+#    {"question":"QUESTION_TEXT","labels":["SENTENCE1", "SENTENCE2", ...]}
+# Example:
+#    {"question":"what country is the grand bahama island in","labels":["contained by", "contained in", "contained"]}
+# The response is JSON object with key "score" and value containing list of scores, one for each label given
+#    {"score":[SCORE1, SCORE2, ...]}
+
+from __future__ import print_function
+from __future__ import division
+
+import importlib
+import subprocess
+import sys
+import tempfile
+from flask import *
+import json
+import csv
+
+from keras.layers.convolutional import MaxPooling1D, AveragePooling1D
+from keras.layers.recurrent import SimpleRNN, GRU, LSTM
+import pysts.embedding as emb
+import pysts.eval as ev
+import pysts.kerasts.blocks as B
+from pysts.kerasts.objectives import ranknet
+
+import anssel_train as anssel_train
+import models  # importlib python3 compatibility requirement
+from nltk.tokenize import word_tokenize
+
+modelname = sys.argv[1]
+trainf = sys.argv[2]
+weightsfile = sys.argv[3]
+if (len(sys.argv) > 4):
+	port = int(sys.argv[4])
+else:
+	port = 5000
+params = sys.argv[5:]
+
+app = Flask(__name__)
+
+def load_jacana(question, prop_labels):
+    samples = []
+    q = word_tokenize(question.replace("?",""))
```

Comment:
Why removing the question mark? (It's probably not wrong, just a little bit strange preprocessing step that we do not do anywhere else I think...)